ambari-server/src/main/java/org/apache/ambari/server/agent/stomp/dto/MetadataCluster.java

```diff
@@ -101,7 +110,16 @@ public boolean updateClusterLevelParams(SortedMap<String, String> update) {
         break;
       }
     }
+
+    for (String key : clusterLevelParams.keySet()) {
+      if (!update.containsKey(key) || !StringUtils.equals(update.get(key), clusterLevelParams.get(key))) {
+        changed = true;
+        break;
+      }
+    }
+
```

Comment:
This method is almost identical to `updateServiceLevelParams`.  (The duplication is not introduced in this change, but is highlighted by the fact that the same problem needs to be fixed in two places.)  Can you please extract a method that takes 2 maps and updates one of them from the other?  Then please delegate to that method from these two.  Please make sure to keep the version that uses `null`-safe comparison.

Reply:
Good idea; I'll extract it to a util method (maybe an existing/new class; will check this out)

Reply:
Done